Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp

```diff
@@ -389,6 +399,91 @@ namespace AzToolsFramework
         }
     }
 
+    ActionManagerOperationResult ActionManager::RegisterWidgetAction(
+        const AZStd::string& widgetActionIdentifier, const WidgetActionProperties& properties, AZStd::function<QWidget*()> generator)
+    {
+        if (m_widgetActions.contains(widgetActionIdentifier))
+        {
+            return AZ::Failure(
+                AZStd::string::format("Action Manager - Could not register widget action \"%.s\" twice.", widgetActionIdentifier.c_str()));
+        }
+
+        m_widgetActions.insert(
+            {
+                widgetActionIdentifier,
+                EditorWidgetAction(
+                  widgetActionIdentifier,
+                  properties.m_name,
+                  properties.m_category,
+                  AZStd::move(generator)
```

Comment:
what happens when you do a mode where its passed by value?

Reply:
It will move the value instead of copying it one more time. It's a minor optimization since we don't use it anymore in this function, and the constructor for EditorWidgetAction takes is by value.